src/mon/OSDMonitor.cc

```diff
@@ -341,11 +341,14 @@ bool is_unmanaged_snap_op_permitted(CephContext* cct,
 
 } // anonymous namespace
 
-void LastEpochClean::Lec::report(ps_t ps, epoch_t last_epoch_clean)
+void LastEpochClean::Lec::report(ps_t ps, epoch_t last_epoch_clean, const size_t pg_num)
 {
   if (epoch_by_pg.size() <= ps) {
     epoch_by_pg.resize(ps + 1, 0);
   }
+  if (epoch_by_pg.size() > pg_num) {
+    epoch_by_pg.resize(pg_num, 0);
+  }
```

Comment:
i think the pg seed of pgs in a pool is a consecutive integer sequence in the range of `[0, pg_num)`. 

i think we can simplify the resizing part in this method, like
```c++
epoch_by_pg.resize(pg_num, 0);
```

1.  no need to compare the size with `ps` or `pg_num`,
2. always call `resize()` unconditionally.

because STL checks for the old size before trying to enlarge or shrink the vector, like:
```c++
if (new_size > size()) {
  // append elements
} else if (new_size < size()) {
  // erase elements
}
```

so this would be a no-op in most all cases.

Reply:
OK that makes sense. I hadn't done that optimization yet in case it might break the next_missing logic during bootstrapping. But reviewing that again, it should be OK.